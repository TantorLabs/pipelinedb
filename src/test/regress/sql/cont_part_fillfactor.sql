drop function if exists list_partitions(name);
create function
  list_partitions(
    baserel name
  )
  returns table (
    parent name, child text, options text[]
  )
  as $BODY$
    begin
      return query select
        parent.relname      as parent,
        regexp_replace(child.relname, 'mrel_[0-9]+_', 'mrel_OID_')       as child,
        child.reloptions as options
      from pg_inherits
        join pg_class parent        on pg_inherits.inhparent = parent.oid
        join pg_class child         on pg_inherits.inhrelid   = child.oid
        join pg_namespace nmsp_parent   on nmsp_parent.oid  = parent.relnamespace
        join pg_namespace nmsp_child    on nmsp_child.oid   = child.relnamespace
      where
        parent.relname = baserel || '_mrel'
      order by child
      ;
    end;
  $BODY$
language plpgsql;

create foreign table part_ff_stream (ts timestamp, x int) server pipelinedb;

create view cv_part_ff
with (partition_by=ts, partition_duration='1 hour')
as select ts, count(*) from part_ff_stream group by ts;

\d+ cv_part_ff_mrel;

insert into part_ff_stream values ('2024-11-12 10:18:42', 1);

select * from list_partitions('cv_part_ff');

insert into part_ff_stream values ('2024-11-12 20:18:42', 2);

select * from list_partitions('cv_part_ff');

drop foreign table part_ff_stream cascade;
