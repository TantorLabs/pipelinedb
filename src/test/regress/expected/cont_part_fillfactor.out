drop function if exists list_partitions(name);
NOTICE:  function list_partitions(name) does not exist, skipping
create function
  list_partitions(
    baserel name
  )
  returns table (
    parent name, child text, options text[]
  )
  as $BODY$
    begin
      return query select
        parent.relname      as parent,
        regexp_replace(child.relname, 'mrel_[0-9]+_', 'mrel_OID_')       as child,
        child.reloptions as options
      from pg_inherits
        join pg_class parent        on pg_inherits.inhparent = parent.oid
        join pg_class child         on pg_inherits.inhrelid   = child.oid
        join pg_namespace nmsp_parent   on nmsp_parent.oid  = parent.relnamespace
        join pg_namespace nmsp_child    on nmsp_child.oid   = child.relnamespace
      where
        parent.relname = baserel || '_mrel'
      order by child
      ;
    end;
  $BODY$
language plpgsql;
create foreign table part_ff_stream (ts timestamp, x int) server pipelinedb;
create view cv_part_ff
with (partition_by=ts, partition_duration='1 hour')
as select ts, count(*) from part_ff_stream group by ts;
\d+ cv_part_ff_mrel;
                                  Partitioned table "public.cv_part_ff_mrel"
 Column |            Type             | Collation | Nullable | Default | Storage | Stats target | Description 
--------+-----------------------------+-----------+----------+---------+---------+--------------+-------------
 ts     | timestamp without time zone |           | not null |         | plain   |              | 
 count  | bigint                      |           |          |         | plain   |              | 
 $pk    | bigint                      |           | not null |         | plain   |              | 
Partition key: RANGE (ts)
Indexes:
    "cv_part_ff_mrel_pkey" PRIMARY KEY, btree ("$pk", ts)
    "cv_part_ff_mrel_expr_idx" btree (pipelinedb.ls_hash_group(ts))
Number of partitions: 0

insert into part_ff_stream values ('2024-11-12 10:18:42', 1);
select * from list_partitions('cv_part_ff');
     parent      |                 child                  |     options     
-----------------+----------------------------------------+-----------------
 cv_part_ff_mrel | mrel_OID_20241112100000_20241112110000 | {fillfactor=50}
(1 row)

insert into part_ff_stream values ('2024-11-12 20:18:42', 2);
select * from list_partitions('cv_part_ff');
     parent      |                 child                  |     options     
-----------------+----------------------------------------+-----------------
 cv_part_ff_mrel | mrel_OID_20241112100000_20241112110000 | {fillfactor=50}
 cv_part_ff_mrel | mrel_OID_20241112200000_20241112210000 | {fillfactor=50}
(2 rows)

drop foreign table part_ff_stream cascade;
NOTICE:  drop cascades to view cv_part_ff
